python/amici/sbml_import.py

```diff
@@ -1539,7 +1539,7 @@ def _add_conservation_for_non_constant_species(
         try:
             sorted_state_exprs = toposort_symbols(state_exprs)
         except CircularDependencyError as e:
-            # see SBML semantic test suite, case 18 for an example
+            # This should never happen
```

Comment:
if this should never happen maybe raise an error instead of logging a warning?

Reply:
Makes sense. Will re-raise.